okhttp/src/main/java/okhttp3/internal/connection/ExchangeFinder.java

```diff
@@ -110,7 +110,7 @@ private RealConnection findHealthyConnection(int connectTimeout, int readTimeout
 
       // If this is a brand new connection, we can skip the extensive health checks.
       synchronized (connectionPool) {
-        if (candidate.successCount == 0) {
+        if (candidate.successCount == 0 && !candidate.isMultiplexed()) {
```

Comment:
This is the biggest difference with the 4.x change. This condition wasn’t necessary there because we increment successCount on a timed out HTTP/2  request. (possibly mistakenly!)